packages/video_player/video_player_android/android/src/main/java/io/flutter/plugins/videoplayer/VideoPlayer.java

```diff
@@ -169,6 +169,27 @@ private MediaSource buildMediaSource(
     }
   }
 
+  public void recreateSurface(Context context) {
+    ExoPlayer exoPlayer = new ExoPlayer.Builder(context).build();
+
+    exoPlayer.setMediaSource(mediaSource);
+    exoPlayer.prepare();
+
+    setUpVideoPlayer(exoPlayer, new QueuingEventSink());
+    exoPlayer.setVideoSurface(surfaceProducer.getSurface());
+  }
+
+  public void pauseSurface() {
+    eventChannel.setStreamHandler(null);
```

Comment:
I agree but I'm not sure because I'm not deeply familiar with ExoPlayer. Does the case where a video is playing and then the app is moved to the background and brought back get impacted by this, i.e. the video is no longer playing from the point it was stopped?

Reply:
Let me investigate this .

Reply:
I looked at this and it seems like the video state is reset. I'm going to see if there is a way to restore it.

Reply:
I added logic to restore all of the visible exo state when re-creating exo player.